src/rules/no-duplicate-definitions.js

```diff
@@ -85,11 +85,11 @@ export default {
 			),
 		);
 
-		/** @type {Set<string>} */
-		const definitions = new Set();
+		/** @type {Map<string, { line: number, label: string }>} */
+		const definitions = new Map();
```

Comment:
```suggestion
		/** @type {Map<string, Definition>} */
		const definitions = new Map();
```

Could we use the existing `Definition` type here?

Using the `Definition` type would allow us to reuse existing object references and make the code structure simpler and clearer.

FYI, the `Definition` and `FootnoteDefinition` types can be imported as follows:

```diff
/**
+* @import { Definition, FootnoteDefinition } from "mdast";
 * @import { MarkdownRuleDefinition } from "../types.js";
 * ...
 ```
 
Maybe the final code would look like this:

<img width="591" height="536" alt="image" src="https://github.com/user-attachments/assets/6b92f1d9-be3c-4a76-a3c6-62c9ab0aa033" />

Reply:
It's okay, we'll only store references to objects that would exist anyway, so there's no perf cost (and there's even a small gain as we are not creating new objects).